### ISSUE TITLE
Idle.setIdleName('yourIdleName') 

### ISSUE THREAD

**Issue by soniyaBisht04** [HackedByChinese/ng2-idle#87]
I want to create multiple instances of idle in the project.
I tried using setIdleName each and every time am creating new instance, still timeout of one idle instance is triggering another timeout subscriber. 

How we can actually make use of setIdleName or check which idle instance we are working on?

**Comment by fetters5** [comment]
I've noticed that using setIdleName results in the appropriate creation of the new storage keys, but also sill created the default idling key
![image](https://user-images.githubusercontent.com/3276557/36866631-d0e5c512-1d60-11e8-93b3-9a806dfe8cc8.png)

Not sure if that could potentially be causing issues for you

**Comment by ednjv** [comment]
@soniyaBisht04  the way I figured to fix this it's to define a provider on each component that is going to declare its own ng2Idle instance. e.g

```
import { Idle, IdleExpiry, LocalStorageExpiry } from '@ng-idle/core';

@Component({
  providers: [LocalStorageExpiry, { provide: IdleExpiry, useExisting: LocalStorageExpiry }, Idle],
  selector: 'component-a',
  template: ''
})
export class ComponentA {
  constructor(public idle: Idle) {
    this.idle.setIdleName('idleB');
  }
}
```


```
import { Idle, IdleExpiry, LocalStorageExpiry } from '@ng-idle/core';

@Component({
  providers: [LocalStorageExpiry, { provide: IdleExpiry, useExisting: LocalStorageExpiry }, Idle],
  selector: 'component-b',
  template: ''
})
export class ComponentB {
  constructor(public idle: Idle) {
    this.idle.setIdleName('idleB');
  }
}
```


I hope this helps

**Comment by jjlorenzo** [comment]
I'm still see the `ng2Idle.main.idling` being created

**Comment by ednjv** [comment]
@jjlorenzo yeah, can't tell right now how to avoid that. What I'm doing is re-using the main idle for whatever I need, and then create another instance for something else

**Comment by jjlorenzo** [comment]
makes sense! thanks for confirming that
